contrib/natss/pkg/dispatcher/dispatcher/dispatcher_test.go

```diff
@@ -106,6 +115,54 @@ func TestSubscribeUnsubscribe(t *testing.T) {
 	}
 }
 
+func TestMalformedMessage(t *testing.T) {
+	logger.Info("TestMalformedMessage()")
+
+	cRef := provisioners.ChannelReference{Namespace: "test_namespace", Name: "test_channel"}
+	sRef := subscriptionReference{Name: "sub_name", Namespace: "sub_namespace", SubscriberURI: "", ReplyURI: ""}
+
+	// subscribe to a channel
+	if _, err := s.subscribe(cRef, sRef); err != nil {
+		t.Errorf("Subscribe to NATSS failed: %v", err)
+	}
+	defer func() {
+		if err := s.unsubscribe(cRef, sRef); err != nil {
+			t.Errorf("Close subscription to NATSS failed: %v", err)
+		}
+	}()
+
+	m := &provisioners.Message{
+		Headers: map[string]string{"header1": "value1", "header2": "value2"},
+		Payload: []byte{'1', '2', '3', '4', '5'},
+	}
+	ch := getSubject(cRef)
+	message, err := json.Marshal(m)
+	if err != nil {
+		t.Errorf("Error during marshaling of the message: %v", err)
+	}
+	// Corrupting message so it would fail Unmarshal
+	b := message[:len(message)-3]
+	if err := stanutil.Publish(s.natssConn, ch, &b, testLogger.Sugar()); err != nil {
+		logger.Errorf("Error during publish: %v", err)
```

Comment:
You shouldn't need the `logger.Errorf` here, because `t.Errorf` should print the error while failing the test.